Automattic/wp-calypso
# Launchpad: Find the best domain to use for iFrame and for sidebar domain container

#### Proposed Changes

We want to find the best domain to use for the iFrame and for the sidebar domain container.

For the iFrame the best domain will always be the `.wordpress.com` because it will have a SSL certificate at all times so it won't cause an error when trying to load the iFrame (as opposed to trying to load a recently purchase domain that does have not have SSL certificates ready thus making it insecure). If the `.wordpress.com` domain doesn't exist for some reason (technically very unlikely but just in case), as a safety fallback we are choosing the most recently registered domain name for that site.

For the sidebar domain container, the best domain will always be the most recently registered domain name for that site (the custom domain when the user has purchased a custom domain) because that will be the domain that the user would expect to see. If there's no custom domain we are choosing the `.wordpress.com` domain

#### Testing Instructions

Custom domains:

1. Start a new newsletter flow or link-in-bio flow and create a site with a custom domain
2. Once you arrive at the Launchpad screen you should see that in the sidebar domain container we see the custom domain (even though you may have the `.wordpress.com` domain in the `siteSlug` param)
3. The iFrame should load correctly because it will use the `.wordpress.com` domain instead of a custom domain (which will most likely be lacking an SSL cert at this point)
4. Start a new newsletter flow or link-in-bio flow WITHOUT creating a custom domain
5. Once you arrive at the Launchpad screen you should see that in the sidebar domain container we see the `.wordpress.com` domain since no custom domain is found
6. Ensure links that navigate away from the Launchpad are redirecting to the correct pages (`Go to Admin` and checklist items)

#### Pre-merge Checklist






Fixes #68519 
Fixes #68488
